src/v/cluster/topics_frontend.cc

```diff
@@ -1331,14 +1327,19 @@ ss::future<std::error_code> topics_frontend::increase_replication_factor(
               return ss::now();
           }
 
-          partition_constraints.partition_id = p_id;
-
           return _allocator
             .invoke_on(
               partition_allocator::shard,
-              [topic,
-               partition_constraints,
-               assignment = std::move(assignment)](partition_allocator& al) {
+              [new_rf = new_replication_factor,
+               topic,
+               disk_max_usage = hard_max_disk_usage_ratio,
+               assignment = std::move(assignment),
+               p_id,
+               h_report](partition_allocator& al) {
+                  auto partition_constraints = get_partition_constraints(
+                    model::partition_id(0), new_rf, disk_max_usage, h_report);
```

Comment:
why `partition_id(0)` though? It should use the size of currently reallocated partition, not partition 0 in constraints. Was this another bug?

Reply:
good point